internal/ingress/controller/store/store.go

```diff
@@ -529,44 +529,39 @@ func New(
 		AddFunc: func(obj interface{}) {
 			cm := obj.(*corev1.ConfigMap)
 			key := k8s.MetaNamespaceKey(cm)
-			// updates to configuration configmaps can trigger an update
-			if key == configmap || key == tcp || key == udp {
```

Comment:
@ElvinEfendi These were the issue - the Store only notified the ingress of updates to a small set of ConfigMaps that are specified using flags when the controller is started (the global, tcp and udp ConfigMaps).

This fixes my issue (the end-to-end tests pass now), but I'm worried about the impact of this change on a noisy cluster (one where there's a lot of ConfigMap changes)? It would definitely have some performance impact. It seems like the alternative would be to rearchitect how the Store works so it supports a whitelist that can be updated on the fly, which is a big and fairly complex, risky change.

Reply:
@actgardner yes, we restrict when to update to avoid too many invocations to Sync(), which could not trigger any change in nginx but increase the CPU and memory utilization